docs/source/developer_guides/lora.md

```diff
@@ -69,6 +69,16 @@ from peft import LoraConfig
 config = LoraConfig(use_rslora=True, ...)
 ```
 
+### Weight-Decomposed Low-Rank Adaptation (DoRA)
+
+This technique decomposes the updates of the weights into two parts, magnitude and direction. Direction is handled by normal LoRA, whereas the magnitude is handled by a separate learnable parameter. This can improve the performance of LoRA, especially at low ranks. Right now, DoRA only supports non-quantized linear layers. DoRA introduces a bigger overhead than pure LoRA, so it is recommended to merge weights for inference. For more information, see  https://arxiv.org/abs/2402.09353.
```

Comment:
If we support conv layers for regular LoRA, would it be very non-trivial to support conv layers with DoRA?

Reply:
I would also hyperlink "merge weights for inference" so that users know what we're talking about here.

Reply:
I think DoRA can definitely be applied on conv2d similar to how LoRA is applied to conv2d. And the normalization could be calculated per-input channel.

Reply:
Thanks! We will loop you in when we add the support! 

Cc: @BenjaminBossan

Reply:
Great. Let's add Conv2d in a follow-up PR once this one is merged.